reco_utils/recommender/sar/sar_singlenode.py

```diff
@@ -94,44 +103,66 @@ def __init__(
         # track user-item pairs seen during training
         self.seen_items = None
 
-    def compute_affinity_matrix(self, df, n_users, n_items):
+    def compute_affinity_matrix(self, df, rating_col):
         """ Affinity matrix
         The user-affinity matrix can be constructed by treating the users and items as
         indices in a sparse matrix, and the events as the data. Here, we're treating
         the ratings as the event weights.  We convert between different sparse-matrix
         formats to de-duplicate user-item pairs, otherwise they will get added up.
         
         Args:
-            df (pd.DataFrame): Indexed df of users and items.
-            n_users (int): Number of users.
-            n_items (int): Number of items.
+            df (pd.DataFrame): Indexed df of users and items
+            rating_col (str): Name of column to use for ratings
 
         Returns:
             sparse.csr: Affinity matrix in Compressed Sparse Row (CSR) format.
         """
 
         return sparse.coo_matrix(
-            (df[self.col_rating], (df[self.col_user_id], df[self.col_item_id])),
-            shape=(n_users, n_items),
+            (df[rating_col], (df[self.col_user_id], df[self.col_item_id])),
+            shape=(self.n_users, self.n_items),
         ).tocsr()
 
-    def compute_coocurrence_matrix(self, df, n_users, n_items):
+    def compute_time_decay(self, df, decay_column):
+        """Compute time decay on provided column
+
+        Args:
+            df (pd.DataFrame): DataFrame of users and items
+            decay_column (str): column to decay
+
+        Returns:
+            DataFrame: with column decayed
+        """
+
+        # if time_now is None use the latest time
+        if not self.time_now:
```

Comment:
here it would be more correct to do: `if self.time_now is None`

Reply:
yep, will do